File: src/main/java/com/logics/logics/repositories/UserRepository.java
package com.logics.logics.repositories;

import com.logics.logics.entities.User;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Mono;

public interface UserRepository extends ReactiveCrudRepository<User, String> {
  Mono<User> findByUsername(String username);
}

File: src/main/java/com/logics/logics/repositories/UserRepository.kt
package com.logics.logics.repositories

import com.logics.logics.entities.User
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface UserRepository : ReactiveCrudRepository<User, String> {
    fun findByUsername(username: String): Mono<User>
    fun findByUsernameContainingIgnoreCase(username: String): Flux<User>

}